### PR TITLE
Created a ENV variable to check if the captcha is enable.

### PR DESCRIPTION
**Pull request by albarnaz**
[#17714](https://central.cm42.io/projects/central#story-17714)

### CODE REVIEW

File: app/controllers/teams_controller.rb [hunk 1]
```diff
@@ -125,7 +125,8 @@ def allowed_params
   end
 
   def can_create?
-    verify_recaptcha && (@team = TeamOperations::Create.call(@team, current_user))
+    binding.pry
```
Comment (dmitryrck) [inline app/controllers/teams_controller.rb]: DEBUG DETECTED

Comment (eddiecode) [inline app/controllers/teams_controller.rb]: I think you forget to remove it 😛

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -86,4 +86,9 @@ def set_layout_settings
     @layout_settings_default = { fluid: false }.freeze
     @layout_settings = @layout_settings_default.dup
   end
+
+  def show_recaptcha?
+    ENV["ENABLED_RECAPTCHA"] || false
```
Comment (dmitryrck) [inline app/controllers/application_controller.rb]: I guess you don't need this `nil` is _falsey_ :P

File: app/controllers/registrations_controller.rb [hunk 3]
```diff
@@ -92,4 +92,8 @@ def set_locale
   def reset_locale
     session[:locale] = nil
   end
+
+  def no_need_check?
+    !show_recaptcha?
```
Comment (dmitryrck) [inline app/controllers/registrations_controller.rb]: :+1:

File: app/controllers/teams_controller.rb [hunk 4]
```diff
@@ -125,7 +125,8 @@ def allowed_params
...
+    check_recaptcha && (@team = TeamOperations::Create.call(@team, current_user))
```
Comment (dmitryrck) [inline app/controllers/teams_controller.rb]: You don't need these parenthesis `@team = …` works :P

File: app/controllers/teams_controller.rb [hunk 5]
```diff
@@ -137,4 +138,8 @@ def add_team_for(user)
       flash[:notice] = t('teams.team_was_successfully_updated')
     end
   end
+
+  def check_recaptcha
+    (show_recaptcha?)? verify_recaptcha : true
```
Comment (dmitryrck) [inline app/controllers/teams_controller.rb]: Is this a ternary? Maybe you can come with something like:

```
show_recaptcha? && verify_recaptcha
```

I'm quite sure this `true` is avoidable 🤔

Comment (dmitryrck) [reply app/controllers/teams_controller.rb]: Besides this expression `(show_recaptcha?)?` is not usual 🤔

File: app/views/application/_captcha.html.erb [hunk 6]
```diff
@@ -0,0 +1,6 @@
+<% field_class = field_class || '' %>
```
Comment (dmitryrck) [inline app/views/application/_captcha.html.erb]: 😐 Don't do this, don't create variables in a view, _and_ you are using it only once :P

Comment (andrerpbts) [reply app/views/application/_captcha.html.erb]: field_class.presence :)

Comment (dmitryrck) [reply app/views/application/_captcha.html.erb]: _And_ if `field_class` is `nil`, erb will do nothing with it :P

**Comment by andrerpbts** [comment]
Add a CHANGELOG entry, please...

File: app/controllers/teams_controller.rb (hunk 1 above)
Comment (albarnaz) [reply app/controllers/teams_controller.rb]: Sorry, already been fixed

File: app/controllers/teams_controller.rb (hunk 5 above)
Comment (patybastos) [reply app/controllers/teams_controller.rb]: Omit parentheses for ternary conditions or use somathing like @dmitryrck said

File: app/controllers/registrations_controller.rb (hunk 3 above)
Comment (andrerpbts) [reply app/controllers/registrations_controller.rb]: Why did you create this method?

File: app/controllers/application_controller.rb [hunk 7]
```diff
@@ -86,4 +86,9 @@ def set_layout_settings
     @layout_settings_default = { fluid: false }.freeze
     @layout_settings = @layout_settings_default.dup
   end
+
+  def show_recaptcha?
+    ENV["ENABLE_RECAPTCHA"] || false
```
Comment (andrerpbts) [inline app/controllers/application_controller.rb]: ENV.fetch("ENABLE_RECAPTCHA", false)

**Pull request merged**